config_utilities/demos/demo_config.cpp

```diff
@@ -160,78 +161,362 @@ void declare_config(SubSubConfig& config) {
+    return 1;
+  }
 
-  // Easier automatic printing of all configs with unit and additional information can be done using the toString():
-  const std::string config_as_string = config::toString(config);
-  std::cout << config_as_string << std::endl;
+  const std::string my_root_path = std::string(argv[1]) + "/";
 
-  // Inclunding "printing.h" also implements the ostream operator for decared config types. The above is thus equivalent
-  // to:
-  std::cout << config << std::endl;
+  // GLobal settings can be set at runtime to change the behavior and presentation of configs.
+  config::Settings().inline_subconfig_field_names = true;
+  config::Settings().print_warnings = true;
+
+  SimpleConfig config;
+  std::cout << config::toString(config) << std::endl;
+  auto interface = std::make_shared<ManualInterface>();
+  DynamicReconfigureServer::registerInterface(interface);
+
+  //I am unclear about this key. Does it need to be namespaced according to anything else in the system?
+  //Ideally not as its just some key that refers to an INSTANCE of a config and not the Config type (like config::name())
+  //some bookeeping could be done here. 
+  DynamicReconfigureServer::registerConfig<ManualInterface>(&config, "my_config");
+
+  SimpleConfig new_config;
+  interface->set(new_config, "my_config");
+
+  new_config.f = 10.0;
+  interface->set(new_config, "my_config");
+
+  std::cout << config::toString(config) << std::endl;
+
+  interface->set(new_config, "non_existant_config");
+
+  //should not work as different config types (between what is registered as "my_config" and the incoming config)
+  demo::SubConfig sub_config;
+  interface->set(sub_config, "my_config");
+
+
+  /**
+   * @brief Ideas
+   * sub config modification via namespacing
+   * a config is registered with a name, e.g "my_config" and (using the MetaData) we make namespacing for each of the fields
+   * e.g my_config/f, my_config/s
+   * 
+   * The interface gets a new function that allows sub-updating e.g. Interface::setField(T, namespace)
+   * e.g interface->setField(10.1, "my_config/f")
+   * 
+   * since we have access to my_config and we know how to access the YAML field 'f' associated with that config we can just update the field in the yaml.
+   * In this way we dont need to check the type of T against the type of SimpleConfig::f, we just need to check it against the type in the YAML. 
+   * Then we can update the new yaml, construt a new SimpleConfig out of it with just the new value for f and update the config as usual.
+   * 
+   * Of course you can update the whole config by using the "base"/(root?) namespace, which in this example would be "my_config". This is also type checked as we store 
+   * the type of SimpleConfig (as is currently implemented)/
+   * 
+   * This would get around the client side needing to know how to build the WHOLE Config type, you just need to know the namespace and the type of the field (which I guess you 
```

Comment:
This was the idea behind the YAML-based interface, i.e. the client won't have to know anything about the config, it will send the updates and the local config will know what to do with it (and ideally send a response in the final version)
If one knows the type of-course it could be nice to provide a templated client that can do some checks locally before sending to the config if this is what you're referring to?